### PR TITLE
use dataclasses for socket events

### PR DESCRIPTION
**Pull request by PythonFZ**
For a better overview of how the data is structured this PR introduces dataclasses.
This is the first step in improving the socket communication structures by analyzing what is currently in use.

### CODE REVIEW

File: zndraw/server/data.py
Comment (RokasEl) [inline zndraw/server/data.py:1]: Could we have a base data class which always contains a token? I see that some objects have a token, some sid but never both. Maybe we could just call it `token` in general? Although that could lead to more confusion as well

File: zndraw/server/events.py [hunk 1]
```diff
@@ -9,8 +10,33 @@
 from flask import request, session
 from flask_socketio import call, emit, join_room
 
+from zndraw.utils import typecast
+
 from ..app import cache
 from ..app import socketio as io
+from .data import (
```
Comment (RokasEl) [inline zndraw/server/events.py:17]: At this point, I'd call the file not `data.py` but `data_transfer_objects.py` and do from `.data_transfer_objects import *`

File: zndraw/server/events.py [hunk 2]
```diff
@@ -19,21 +45,35 @@
 
 def _webclients_room(data: dict) -> str:
     """Return the room name for the webclients."""
-    if "sid" in data:
-        return data["sid"]
-    return f"webclients_{data['token']}"
+    if isinstance(data, dict):
```
Comment (RokasEl) [inline zndraw/server/events.py:48]: If we had a base class, we could implement a custom `__contains__` method that removes this boilerplate.

File: zndraw/server/events.py [hunk 3]
```diff
@@ -19,21 +45,35 @@
...
     """Return the SID of the default webclient."""
-    if "sid" in data:
-        return data["sid"]
-    # TODO: if there is a keyerror, it will not be properly handled and the
-    #  python interface is doomed to wait for TimeoutError.
-    try:
-        return f"webclients_{data['token']}"
-    except KeyError:
-        log.critical("No webclient connected.")
+    if isinstance(data, dict):
```
Comment (RokasEl) [inline zndraw/server/events.py:60]: see above

File: zndraw/server/events.py [hunk 4]
```diff
@@ -43,8 +83,12 @@ def _pyclients_room(data: dict) -> str:
 
 def _pyclients_default(data: dict) -> str:
     """Return the SID of the default pyclient."""
-    if "sid" in data:
-        return data["sid"]
+    if isinstance(data, dict):
```
Comment (RokasEl) [inline zndraw/server/events.py:86]: see above

File: zndraw/server/events.py [hunk 5]
```diff
@@ -300,7 +344,9 @@ class Scene(BaseModel):
 
 
 @io.on("modifier:run")
-def modifier_run(data):
+@typecast
+def modifier_run(data: ModifierRunData):
+    print(data)
```
Comment (RokasEl) [inline zndraw/server/events.py:349]: We should either use `logging` or remove prints outside debugging

**Review by RokasEl — CHANGES_REQUESTED** [review]

File: zndraw/server/events.py (hunk 1 above)
Comment (PythonFZ) [reply zndraw/server/events.py:17]: The number of imports will be dramatically reduced in the future, so I'd like to keep this - seeing the number of imports being reduced also acts as motivation.

Also, I don't like wildcard imports.

**Pull request merged**